deploy/terraform/terraform-units/modules/sap_landscape/storage_accounts.tf

```
@@ -576,7 +573,7 @@ resource "azurerm_private_endpoint" "install" {
                                            azurerm_storage_share.install,
                                            azurerm_storage_share.install_smb
                                          ]
-  count                                = local.use_AFS_for_shared && var.use_private_endpoint ? (
+  count                                = local.use_AFS_for_shared && var.use_private_endpoint && var.infrastructure.virtual_networks.sap.subnet_app.defined ? (
```

We cannot use the .defined here as it would break if the customer provides the Azure Resource Id for the app subnet. 
```suggestion
  count                                = local.use_AFS_for_shared && var.use_private_endpoint ? (
```